### PR TITLE
AzureMonitor: Azure Monitor Cheat sheet

### LINKED ISSUE #72115: Implement cheatsheet for Azure Monitor Logs
Add a cheatsheet for Azure Monitor Logs that helps users kick start their queries. A list of queries to include can be found in the Azure documentation: https://learn.microsoft.com/en-us/azure/azure-monitor/logs/get-started-queries.

### PR DESCRIPTION
**Pull request by alyssajoyner**
Adds a cheatsheet to Azure Monitor Logs queries.

https://github.com/grafana/grafana/assets/58453566/0d1d0c7b-02b9-4b45-a8ef-2bb7c523a472

Closes #72115

### CODE REVIEW

File: public/app/plugins/datasource/azuremonitor/components/AzureCheatSheet.tsx [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+  CustomScrollbar,
+  InlineField,
+  Input,
+  LoadingPlaceholder,
+  Select,
+  useStyles2,
+} from '@grafana/ui';
+
+import tokenizer from '../../cloudwatch/language/cloudwatch-logs/syntax';
+import { RawQuery } from '../../prometheus/querybuilder/shared/RawQuery';
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/components/AzureCheatSheet.tsx:19]: We don't want to be importing from other areas of Grafana core, are these components functions that we can bring into Azure Monitor easily?

File: public/app/plugins/datasource/azuremonitor/components/AzureCheatSheet.tsx [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+        return {
+          label: category,
+          value: category,
+        };
+      });
+    }
+    return [];
+  }, [cheatsheetQueries]);
+
+  const getCheatsheetQueries = async () => {
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/components/AzureCheatSheet.tsx:85]: Can we move this function up to the datasource and then add the request to the backend datasource?

File: public/app/plugins/datasource/azuremonitor/components/AzureCheatSheet.tsx [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+                value={searchInputValue}
+                onChange={(e) => {
+                  setSearchInputValue(e.currentTarget.value);
+                  const filteredQueries = filterQueriesBySearch(e.currentTarget.value);
+                  setVisibleQueries(filteredQueries);
+                }}
+                placeholder="Search Logs queries"
+                width={50}
+              />
+              <InlineField label="Categories" grow={true} labelWidth="auto" className={styles.categoryDropdown}>
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/components/AzureCheatSheet.tsx:187]: Can we use `Field` instead of `InlineField`?

**Review by aangelisc — CHANGES_REQUESTED** [review]
Great work on this @alyssabull! It looks amazing 😄 I've got some comments/questions and I think this is missing some testing (and potentially tracking). Otherwise, I think it's a great addition to the query editor that'll really help users ❤️

File: public/app/plugins/datasource/azuremonitor/components/AzureCheatSheet.tsx (hunk 1 above)
Comment (alyssabull) [reply public/app/plugins/datasource/azuremonitor/components/AzureCheatSheet.tsx:19]: Yes good catch.

**Review by aangelisc — APPROVED** [review]
Excellent work! 💯

File: public/app/plugins/datasource/azuremonitor/azure_log_analytics/azure_log_analytics_datasource.ts [hunk 4]
```diff
@@ -252,4 +252,8 @@ export default class AzureLogAnalyticsDatasource extends DataSourceWithBackend<
   private isValidConfigField(field: string | undefined): boolean {
     return typeof field === 'string' && field.length > 0;
   }
+
+  async getAzureLogAnalyticsCheatsheetQueries() {
+    return await this.getResource(`${routeNames.logAnalytics}/v1/metadata`);
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/azure_log_analytics/azure_log_analytics_datasource.ts:257]:
```suggestion
    return await this.getResource(`${this.resourcePath}/v1/metadata`);
```

File: public/app/plugins/datasource/azuremonitor/components/syntax.ts [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+    alias: 'function',
+  },
+  function: {
+    pattern: new RegExp(`\\b(?:${FUNCTIONS.map((f) => f.label).join('|')})\\b`, 'i'),
+  },
+  keyword: {
+    pattern: new RegExp(`(\\s+)(${KEYWORDS.join('|')})(?=\\s+)`, 'i'),
+    lookbehind: true,
+  },
+  // 'log-group-name': {
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/components/syntax.ts:347]: Unneeded comment?

File: public/app/plugins/datasource/azuremonitor/utils/common.ts [hunk 6]
```diff
@@ -43,6 +43,7 @@ export const routeNames = {
   logAnalytics: 'loganalytics',
   appInsights: 'appinsights',
   resourceGraph: 'resourcegraph',
+  logsCheatSheet: 'logscheatsheet',
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/utils/common.ts:46]: I think this is unneeded now?

Comment (alyssabull) [reply public/app/plugins/datasource/azuremonitor/utils/common.ts:46]: Good catch - ty

**Review by bossinc — APPROVED** [review]
🎉

**Pull request merged**